src/organization/screens/organization-profile.screen.js

```diff
@@ -109,83 +87,82 @@ class OrganizationProfile extends Component {
 
   handleActionSheetPress = index => {
     if (index === 0) {
-      openURLInView(this.props.organization.html_url);
+      openURLInView(this.props.org._entityUrl);
     }
   };
 
   render() {
     const {
-      organization,
+      orgId,
+      org,
       members,
-      isPendingOrg,
-      isPendingMembers,
+      membersPagination,
       navigation,
       language,
     } = this.props;
     const { refreshing } = this.state;
     const initialOrganization = this.props.navigation.state.params.organization;
     const organizationActions = [translate('common.openInBrowser', language)];
 
+    if (!org) {
+      return (
+        <Text>
+          Loading organization {orgId} .. TODO: Make me look nicer
+        </Text>
+      );
+    }
+
     return (
       <ViewContainer>
         <ParallaxScroll
-          renderContent={() => (
-            <UserProfile
-              type="org"
-              initialUser={initialOrganization}
-              user={
-                initialOrganization.login === organization.login
-                  ? organization
+          renderContent={() =>
+            <OrgProfile
+              initialOrg={initialOrganization}
+              org={
+                initialOrganization.login === org.login
+                  ? org
                   : initialOrganization
               }
               navigation={navigation}
-            />
-          )}
+            />}
           refreshControl={
             <RefreshControl
-              onRefresh={this.getOrgData}
+              onRefresh={() => this.refreshData()}
               refreshing={refreshing}
             />
           }
-          stickyTitle={organization.name}
+          stickyTitle={orgId}
           navigateBack
           navigation={navigation}
           showMenu
           menuAction={() => this.showMenuActionSheet()}
         >
-          {isPendingMembers && (
+          {membersPagination.isFetching &&
+            !membersPagination.pageCount &&
             <LoadingMembersList
               title={translate('organization.main.membersTitle', language)}
-            />
-          )}
+            />}
 
-          {!isPendingMembers && (
-            <MembersList
+          {(!membersPagination.isFetching || membersPagination.pageCount > 0) &&
```

Comment:
This pagination test should be extracted to a helper function, that tells if a paginator already have done an initial load